src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CMDLineProgramModule.java

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+/**
+ * A class for experimenting with cmd line programs (such as bwa and sga) that's not part of GATK (or has Java bindings yet).
+ */
+public abstract class CMDLineProgramModule {
+
+    public CMDLineProgramModule(){  }
+
+    /**
+     * Builds the initial argument list for the command line (e.g. "bwa mem").
+     * @return a list of strings having the names of requested tools.
+     * @throws IOException
+     */
+    abstract ArrayList<String> initializeCommands() throws IOException;
+
+    /**
+     * Worker function to call the cmd line program.
+     * @param runtimeArguments          arguments to be provided to the program
+     * @param directoryToWorkIn         a directory for the program to work in (mainly for output)
+     * @param stdoutDestination         redirects the stdout from the program to a file
+     * @param stderrMessageDestination  appends the stderr message to a string, which is assumed to be not null
+     * @throws IOException
+     * @throws InterruptedException
+     * @throws RuntimeException
+     */
+    public void run(final ArrayList<String> runtimeArguments,
+                    final File directoryToWorkIn,
+                    final File stdoutDestination,
+                    String stderrMessageDestination) throws IOException, InterruptedException, RuntimeException{
```

Comment:
I'm a little confused about stderrMessageDestination. It looks like this is being used to pass in the first part of an error message (later you seem to append the redirected stderr from the process to it). Why not just make this parameter be "errorMessagePrefix" or something, and then you can just combine it with the actual error message below. That way it can be final like the rest of the parameters to this method, etc., and it would be more understandable.